### PR TITLE
Add new "Lava Puddles" map by smugler5

### PR DESCRIPTION
**Pull request by Avyukt1**
Lava Puddles by smugler5

### CODE REVIEW

**Comment by JostP** [comment]
I reviewed your map. Here are my thoughts:

1. First add Big ocean map back. This PR removes it.

2. One base mustn't have advantage. Currently red base has a lava ring around it and players can dig dirt under their base so lava flow under it and protect it. Other side has lava enclosed with indestructible bricks what causes players cant modify area for their needs. I would like to see both bases have a lava ring around base and indestructible bricks around lava removed.

Zaslonska slika 2020-12-05 12-53-35

Zaslonska slika 2020-12-05 12-53-51

3. There is snow around red base which is somehow not really suitable with lava. It is your choice to remove it or not as it is not that big issue.

Zaslonska slika 2020-12-05 12-54-15

4. There are only diamonds on the map. I am not really sure this is a good idea because pro players will have diamond swords and kill/farm noobs. And we already have a lot of maps with diamond swords. I think it is better idea to add 30 mese on each side. Since knights get only steel swords, one team will be able to collect all mese swords and consequently capture the flag easily as enemy team will be defeated with the loss of all their swords.

File: lava_puddles/map.conf [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+rotation = z
+name = Lava Puddles
+author = smugler5
+h = 140
+team.1.color = red
+team.2.color = blue
+team.2.pos = (101,1,-102)
+team.1 = red
+licence = CC BY-SA 4.0
+hint = Find underground hole to the treaure. Be careful while making tunnel, don't let lava come in your tunnel.
```
Comment (JostP) [inline lava_puddles/map.conf:14]:
```suggestion
hint = Find underground hole to the treaure. Get an advantage by collecting all enemy swords!
```

**Comment by JostP** [comment]
One more thing, you forgot to fill some puddles with lava

Zaslonska slika 2020-12-05 13-25-55

**Comment by Avyukt1** [comment]
I aint able to change it, As I test it, Nothing come in output and I start drowning, the ecported data is not working. Its the problem of the sowtware.

**Comment by Avyukt1** [comment]
Wait, I am making the changes!

**Comment by JostP** [comment]
You can edit exported maps without map maker software:
https://github.com/MT-CTF/capturetheflag/tree/master/mods/ctf/ctf_map/map_maker#editing-exported-map

**Comment by Avyukt1** [comment]
Can you tell me the initial X, Y and Z of the the map?

**Comment by Avyukt1** [comment]
screenshot_20201205_192022

**Comment by Avyukt1** [comment]
I am not able to place the shematic,I will try editting it using the Minetest Schematic Editor (MTSEdit).

**Comment by Avyukt1** [comment]
Nah, Its working now, I thought I had to add .mts to the command.

**Comment by Avyukt1** [comment]
I have made the changes, I will export and redo it tomorrow on GitHub. Good Night

**Comment by Avyukt1** [comment]
Done with fixing bugs.

File: big_ocean/map.conf [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-h = 140
-rotation = x
-team.1 = red
-team.1.color = red
-team.1.pos = (-1,-9,100)
-team.2 = blue
-team.2.color = blue
-team.2.pos = (-1,-9,-100)
-initial_stuff = default:pick_steel,shooter:pistol,default:torch 5
-license = CC BY-SA 4.0
```
Comment (JostP) [inline big_ocean/map.conf:13]: Try to somehow remove this file changes from this PR. I will final review your map today or tomorrow.

Comment (Avyukt1) [reply big_ocean/map.conf:13]: Idk, but the files are same and I tried but its not happening.

**Comment by Avyukt1** [comment]
I've added mese :)

**Comment by Avyukt1** [comment]
@JostP The map has a small bug regarding the barrier. As the match starts, the barrier is not disappearing, It remains the same. Can Can I make changes in the map.mts by changing the position of the barrier? How will I be able to change the barrier even using WorldEdit? 
`OR`
Do we have to make changes in the map.conf file?

**Comment by JostP** [comment]
So your red barrier does not disappear?
Did you edit map.mts file with WorldEdit's //mtschemplace?
When you do //mtschem create, be sure you have placed //1 and //2 **exactly** in the corners of the map.
Try to not make changes to red barrier because you will get errors when playing map in CTF.

**Comment by Avyukt1** [comment]
Bug Fixed

**Comment by JostP** [comment]
Nice work with fixing barrier, I will review your map today

**Comment by Avyukt1** [comment]
@JostP  
Both the sides have the Lava Rings.
One is ahead of the puddles while the other is behind the puddles.

**Comment by JostP** [comment]
You will have to create new PR, because this one touches big ocean map.
So you decided to have diamonds on the map? It is your choice so it is ok.
You will have do make small fix:

Please remove this pillar, but you can leave top node if you need it for export:

Zaslonska slika 2020-12-08 19-32-12

You have to place //1 or //2 like the picture shows. I guess you didn't and as a result bases are one block higher.
Zaslonska slika 2020-12-08 19-35-50

Zaslonska slika 2020-12-08 19-35-57


And finally use this screenshot that I made for you:
Zaslonska slika 2020-12-08 19-30-54

Everything else look fine, nice work :+1:

**Comment by Avyukt1** [comment]
Okay, I'll create a new PR for this map

**Pull request closed without merging**